benchmark/benchmark.py

```diff
@@ -0,0 +1,534 @@
+        )
+
+    def same(self, o: "RunData") -> bool:
+        return all([
+            self.model == o.model,
+            self.parallel == o.parallel,
+            self.batch_size == o.batch_size,
+            self.gpu_count == o.gpu_count,
+            self.gpu_model == o.gpu_model,
+        ])
+
+    def gt(self, o: "RunData") -> bool:
+        return self.end_time > o.end_time
+
+
+@dataclasses.dataclass
+class RunResult:
+    start_time: datetime.datetime
+    end_time: datetime.datetime
+    loss: float
+    accuracy: float
+
+
+BATCH_SIZE_VALUES = (1, 2, 4, 8, 16, 32, 64)
+
+# https://matplotlib.org/stable/gallery/lines_bars_and_markers/marker_reference.html
+# https://matplotlib.org/stable/gallery/lines_bars_and_markers/linestyles.html
+PLOT_LINES = [
+    "solid",
+    "dashed",
+    "dotted",
+    "dashdot",
+]
+PLOT_MARKERS = [
+    "x",
+    "+",
+    ".",
+    "*",
+    "H",
+    "s",
+]
+PLOT_IMG_WIDTH = 10
+PLOT_IMG_HEIGHT = 5
+PLOT_IMG_MARGIN_LEFT = 0.1
+
+
+def gen_plot_lines():
+    for e in PLOT_LINES:
+        yield e
+
+
+def gen_plot_markers():
+    for e in PLOT_MARKERS:
+        yield e
+
+
+def device_name(model: str, count: int) -> str:
+    return f"{count} {model}"
+
+
+def device_filename(model: str, count: int) -> str:
+    return f"{model} {count}".lower().replace(" ", "-")
+
+
+def nofn(args):
+    return 0, 0
+
+
+def wrap(fn, args, q):
+    start_time = datetime.datetime.now()
+    loss, accuracy = fn(args)
+    end_time = datetime.datetime.now()
+    q.put(RunResult(start_time, end_time, loss, accuracy))
+
+
+def identity(string):
+    return string
+
+
+def run_training(model: Models, parallel: Parallelism, batch_size: int, epochs: int, device_name: str, device_count: int, device_model: str) -> RunData:
+
+    args = argparse.ArgumentParser()
+    args.register("type", None, identity)
+    fn = nofn
+
+    if model is Models.none:
+        pass
+    elif model in (Models.orbit5k, Models.orbit5kbig):
+        if model is Models.orbit5kbig:
+            batch_size = 4
+            args.big_model = True
+        else:
+            args.big_model = False
+        args.batch_size = batch_size
+        args.n_epochs = epochs
+        args.parallel = parallel.to_parallelism_type()
+        args.sharding = parallel.to_sharding_strategy()
+        fn = parallel_orbit_5k.main
+    elif model in (Models.bert, Models.bertbig):
+        if model is Models.bertbig:
+            args.big_model = True
+        else:
+            args.big_model = False
+        args.batch_size = batch_size
+        args.n_epochs = epochs
+        args.parallel = parallel.to_parallelism_type()
+        args.sharding = parallel.to_sharding_strategy()
+        args.download = False
+        parallel_bert.download_dataset()
+        fn = parallel_bert.main
+
+    sys.stdout.write(f"++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n")
```

Comment:
Consider using the `logging` module instead to control the logging level and destination better.